lib/include/types/iot_taskpool_types.h

```diff
@@ -191,14 +191,58 @@ typedef enum IotTaskPoolJobStatus
     IOT_TASKPOOL_STATUS_UNDEFINED,
 } IotTaskPoolJobStatus_t;
 
+/*------------------------- Task pool types and handles --------------------------*/
+
+/**
+ * @ingroup taskpool_datatypes_handles
+ * @brief Opaque handle of a Task Pool instance.
+ *
+ * This type identifies a Task Pool instance, which is valid after a successful call
+ * to @ref taskpool_function_createsystemtaskpool or @ref taskpool_function_create. A
+ *  variable of this type is passed as the first
+ * argument to [Task Pool library functions](@ref taskpool_functions) to identify which
+ * task pool that function acts on.
+ *
+ * A call to @ref taskpool_function_destroy makes a task pool handle invalid. Once
+ * @ref taskpool_function_destroy returns, the task handle should no longer
+ * be used.
+ *
+ * @initializer{IotTaskPool_t,IOT_TASKPOOL_INITIALIZER}
+ */
+typedef struct _taskPool * IotTaskPool_t;
+
+/**
+ * @ingroup taskpool_datatypes_structs
+ * @brief The job storage data structure provides the storage for a statically allocated Task Pool Job instance.
+ *
+ * @warning This is a system-level data type that should not be modified or used directly in any application.
+ * @warning This is a system-level data type that can and will change across different versions of the platform, with no regards for backward compatibility.
+ *
+ */
+typedef struct IotTaskPoolJobStorage
+{
+    void * dummy1[ 2 ];             /**< @brief Placeholder. */
```

Comment:
Should probably use the same types (instead of void*) to avoid alignment differences.

Reply:
It is the same type, because it is a pointer. That is all we need to determine the size.